chemprop/v2/data/datasets.py

```diff
@@ -140,6 +140,11 @@ class MoleculeDataset(Dataset, _MolGraphDatasetMixin):
     featurizer: MoleculeMolGraphFeaturizerProto = field(default_factory=MoleculeMolGraphFeaturizer)
 
     def __post_init__(self):
+        if self.data is None:
+            raise ValueError("Data cannot be None!")
```

Comment:
Thanks for the comment, David! There is a unit test [`test_none`](https://github.com/chemprop/chemprop/blob/v2/dev/chemprop/v2/tests/unit/data/test_dataset.py#L67) that expects a `ValueError` when the input for mol is `None`. Before I add this part, this unit test would fail because no exception is raised. That's why I added this to raise the exception. I am happy to remove this part and the `test_none` unit test. Let me know your thoughts!